RELEASE_CHECKLIST.md

```diff
@@ -1,19 +1,42 @@
-1. `git tag` and ensure the name in the list added looks correct, then push the tag directly to the repository by `git push --follow-tags`.
-1. Wait for the completion of the GitHub action [release-github](https://github.com/notaryproject/notation/actions/workflows/release-github.yml).
-1. Check the new draft release, revise the release description, and publish the release.
-1. Announce the release in the community.
+# Release Checklist
+
+## Overview
+
+This document describes the checklist to publish a release via GitHub workflow.
+
+## Release Process
+To release Notation CLI, you might need to do release for [notation-go](https://github.com/notaryproject/notation-go) and [notation-core-go](https://github.com/notaryproject/notation-core-go) libraries.
+
+### [notation-core-go](https://github.com/notaryproject/notation-core-go) release
+
+1. Go to the [releases](https://github.com/notaryproject/notation-core-go/releases) page, and click on `Draft a new release`.
+1. Wait for the PR merge.
+1. Choose/create a tag and in the `Target` dropdown select an appropriate commit. For the release title, reuse the tag name.
+1. Use the `Generate release notes` button to automatically generate release notes.
+1. Save it as a draft and have a peer review it.
+1. Once a peer has reviewed the draft, release the draft by clicking on `Publish Release`.
+
+
+### [notation-go](https://github.com/notaryproject/notation-go) release
+
+1. Bump up the `notation-core-go` dependency in [go.mod](https://github.com/notaryproject/notation-go/blob/main/go.mod) and [go.sum](https://github.com/notaryproject/notation-go/blob/main/go.sum) and open a PR for the changes.
+1. Wait for the PR merge.
+1. Go to the [releases](https://github.com/notaryproject/notation-go/releases) page, and click on `Draft a new release`.
+1. Choose/create a tag and in the `Target` dropdown select an appropriate commit. For the release title, reuse the tag name.
+1. Use the `Generate release notes` button to automatically generate release notes.
+1. Save it as a draft and have a peer review it.
+1. Once a peer has reviewed the draft, release the draft by clicking on `Publish Release`.
+
+
+### [notation](https://github.com/notaryproject/notation) release
+
+1. Determine a [SemVer2](https://semver.org/)-valid version prefixed with the letter `v` for release. For example, `version="v1.0.0-alpha.1"`.
+1. Bump up the `notation-core-go` and `notation-go` dependency in [go.mod](https://github.com/notaryproject/notation-go/blob/main/go.mod) and [go.sum](https://github.com/notaryproject/notation-go/blob/main/go.sum) and open a PR for the changes.
+1. Bump up the `Version` in [internal/version/version.go](internal/version/version.go#L5) and open a PR for the changes.
+1. Wait for the above PR merge.
+1. Be on the main branch connected to the actual repository (not a fork) and `git pull`.  Ensure `git log -1` shows the latest commit on the main branch.
+1. Create a tag `git tag -am $version $version`
```

Comment:
The tag need to be signed now, so the command should be `git tag -sam $version $version`